### DISCUSSION TITLE
Feature condition support in managed UI

### DISCUSSION THREAD

**Discussion by DarkenSral** [oleg-shilo/wixsharp#1430] [Ideas]
Hi Oleg!
I have been looking into getting feature conditions to work also for the managed UI features dialog. Given that the necessary MSI properties used in the feature condition expressions are set up in event handlers for UIIntialized (visible UI) and Load (invisible UI), the commented code changes in FeatureItem.cs below does the trick.
It works fine using the default install levels, and also when assigning a default INSTALLLEVEL other than 1, using non-default Levels on the features and Level on conditions.
I'm putting this suggestion here first get your opinion, before trying to create a PR.

Regards,
Lars

```
public FeatureItem(Session session, string name)
{
	// Debug.Assert(false);

	//
	// Get default install level and feature condition (if any).
	//
	if (!int.TryParse(session["INSTALLLEVEL"], out var installLevel)) installLevel = 1; // MSI default
	var condition = session.OpenView("select * from Condition where Feature_ = '" + name + "'");
	Dictionary<string, object> conditionRow = condition.FirstOrDefault();

	var data = session.OpenView("select * from Feature where Feature = '" + name + "'");
	Dictionary<string, object> row = data.FirstOrDefault();

	if (row != null)
	{
		Name = name;
		ParentName = (string)row["Feature_Parent"];
		Title = (string)row["Title"];
		Description = (string)row["Description"];

		RawDisplay = (int)row["Display"];
		Display = RawDisplay.MapToFeatureDisplay();

		//
		// Set defaultState according to feature and install levels, then evaluate
		// and adjust state according to feature condition.
		//
		var defaultState = (Convert.ToInt32(row["Level"]) <= installLevel) ? InstallState.Local : InstallState.Absent;
		if (session.IsInstalling() 
			&& conditionRow?["Condition"] != null
			&& session.EvaluateCondition(conditionRow["Condition"].ToString())  // If condition is true...
			&& Convert.ToInt32(conditionRow["Level"]) <= installLevel)          // ...set state according to condition level.
		{
			defaultState = InstallState.Local;
		}

		CurrentState = DetectFeatureState(session, name);
		RequestedState = session.IsInstalled() ? CurrentState : defaultState;

		Attributes = (FeatureAttributes)row["Attributes"];
	}
}

```

**Comment by oleg-shilo** [comment]
Great, give me a day or two and I will come back to you on this one. But it seems like a great enhancement.

**Comment by oleg-shilo** [comment]
All good. The change makes perfect sense. A great improvement. 
Please create the PR when you have the time for the. No pressure.

There is only one point to be aware of. 
Please do two separate PRs against both codebase streams: `wix-v3-master` and `wix-v4-master`.
I hope it's not too much trouble.

**Reply by DarkenSral** [reply]
Great, I will fix it today.

**Reply by oleg-shilo** [reply]
Thank you. All merged now
